web/client/components/widgets/enhancers/dependenciesToZoomTo.js

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2023, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import React, { useEffect, memo } from 'react';
+
+const dependenciesToZoomTo = (WrappedComponent)=>{
+    function DependenciesToZoomTo(props) {
+        let tblWidgetWithExtentObj = props.widgets.find(i=>i?.dependencies?.extentObj);
+        const extentObj = tblWidgetWithExtentObj?.dependencies?.extentObj;
+        let mapWidget = props?.widgets?.find(i=>tblWidgetWithExtentObj?.dependenciesMap?.mapSync.includes(i.id));
+        useEffect(()=>{
+            if ( extentObj && mapWidget ) {
+                let connectedMap = mapWidget?.maps.find(i=>i.mapStateSource === mapWidget.id);
+                let zoomToExtentHandler = connectedMap?.zoomToExtentHandler;
+                if (zoomToExtentHandler) {
+                    // trigger "internal" zoom to extent
```

Comment:
normally this is done through map hooks, I'm not sure this is the ideal way to proceed. 
I'll sync with the team @offtherailz @allyoucanmap to hear their opinion too.

Reply:
I think I have tried using the map hooks but in case of more than map and more than table, it doesn't work properly. It zoomed into one map even if this map is not the requested

Reply:
@MV88 @mahmoudadel54 let's wait for @offtherailz feedback before proceeding further.

Reply:
I see we already have a `dependenciesToExtent` that zooms the map, checking the quickFilters, that uses the `hook` logic.  
I think that in any case we should centralize of zoom functionality in that enhancer trying to reuse the logic inside it, so we can concentrate the effort of the feature in one point and avoid duplication of functionalties. 
In case the hook logic do not work in some cases, we may have solve the issue for all the involved use cases in that place. 
So please try to include this your changes in the `dependenciesToExtent` enhancer. 
If some issues comes out figure out the reason and ask. We may solve the issue with hooks or implement an alternative solution that works in all the cases.